tests/CMakeLists.txt

```diff
@@ -264,6 +265,15 @@ foreach(target ${test_targets})
   if(MSVC)
     target_compile_options(${target} PRIVATE /utf-8)
   endif()
+  if (${CMAKE_CXX_COMPILER_ID} MATCHES "PGI")
+    if (WIN32)
+      target_compile_options(${target} PRIVATE -Wc,--pending_instantiations=0 )
+    else()
+      target_compile_options(${target} PRIVATE -Wc,--pending_instantiations=0 -D__GCC_ATOMIC_TEST_AND_SET_TRUEVAL=1)
```

Comment:
Are these always required, or just for the tests? If always, maybe they should go into pybind11Common for pybind11 or some similar target.

Reply:
``-Wc,--pending_instantiations=0`` is about the recursion level in templates. If I recall that correctly.
So, well, in some codes it will not be needed. But in the others the compilation will fail without this option.
Note that this option makes compilation significantly slower.

`-D__GCC_ATOMIC_TEST_AND_SET_TRUEVAL=1` was deduced from experiments with old compilers on Linux.